src/actions/useGetPosts.tsx

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from 'react';
+import { db } from '../firebase';
+import { StateAble } from '../shared/interfaces/StateAbleInterface';
+import { FeedPostProps } from '../components/FeedPost/FeedPost';
+
+export const useGetPosts = (): StateAble<FeedPostProps[]> => {
+  const [state, setState] = useState<StateAble<FeedPostProps[]>>({
+    data: [],
+    isLoading: true,
+    isError: false
+  });
+
+  const getPosts = async () => {
+    try {
+      const snapshot = await db.collection('posts').orderBy('timestamp', 'desc').get();
+      const data = snapshot.docs.map(post => ({ ...post.data(), testid: post.id })) as FeedPostProps[];
+      setState({ ...state, data, isLoading: false });
+    } catch (error) {
+      setState({ ...state, isError: true, isLoading: false });
+    }
+  };
+
+  useEffect(() => {
+    getPosts();
```

Comment:
why you need the function for only one use case

Reply:
if it has to stay like that then keep it but change it if you can

Reply:
I think it improves readability. getPosts is more descriptive for me than the code without the function + in future it's better to move the function somewhere else

Reply:
I think it is easier to read and if we have getPost function there is an opportunity  to reuse it